src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/CourseActivityBuilder.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AgencyInternalLocation
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AgencyLocation
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.IEPLevel
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.ProgramService
+import java.time.LocalDate
+
+class CourseActivityBuilder(
+  var code: String = "CA",
+  var programId: Long = 20,
+  var prisonId: String = "LEI",
+  var description: String = "test course activity",
+  var capacity: Int = 23,
+  var active: Boolean = true,
+  var startDate: String = "2022-10-31",
+  var endDate: String = "2022-11-30",
+  var minimumIncentiveLevelCode: String = "STD",
+  var internalLocationId: Long = -8,
+  var payRates: List<CourseActivityPayRateBuilder> = listOf(CourseActivityPayRateBuilder()),
+) {
+  fun build(
+    prison: AgencyLocation,
+    programService: ProgramService,
+    minimumIepLevel: IEPLevel,
+    internalLocation: AgencyInternalLocation
+  ): CourseActivity {
+    assert(prison.id == this.prisonId)
+    assert(programService.programId == programId)
+    assert(minimumIepLevel.code == minimumIncentiveLevelCode)
+    assert(internalLocation.locationId == internalLocationId)
+    return CourseActivity(
```

Comment:
As we can't retrieve related entities in this class at least check they are the entities we expected.